src/gr11review/part2/Utility.java

```diff
@@ -1,8 +1,384 @@
 package gr11review.part2;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.*;
+import java.util.*;
 
+/**
+ * description: a code to run methods to pass tests
+ * author: @Jenusan, @nathanwan
+ */
+
+/**
+* Where all the methods are located. 
+* @author: N. Wan 
+*
+*/
 public class Utility {
 
+
+    /** A method that checks if "xyz" is in the middle of the string.
+    * @author N. Wan
+    * @param str The string that is going to be checked if there is "xyz" in the middle.
+    * @return boolean If "xyz" is in the middle or not. 
+    */
+    public static boolean xyzMiddle(String str){ 
+
+        // Initialize variables
+        int intLength = str.length();
+        int intSides;
+        String strTester;
+        String strChecker = "xyz";
+
+        // Determine if it is atleast 5 characters long
+        if (intLength >= 5) {
+
+            intSides = intLength - 3;
+            
+            // Check if it is even or odd then check if "xyz" is in the middle.
+            if (intSides % 2 != 0) {
+                
+                intSides = intSides / 2;
     
+                strTester = str.substring(intSides, intSides + 3);
+
+                if (strTester.equals(strChecker) == true) {
+                    
+                    return true;
+
+                } else {
+
+                    strTester = str.substring(intSides +1, intSides + 4);
+
+                    if (strTester.equals(strChecker) == true) {
+                        
+                        return true;
+
+                    }
+                }
+
+            // Check if "xyz" is in the middle for even. 
+            } else {
+
+                intSides = intSides / 2;
+
+                strTester = str.substring(intSides, intSides + 3);
+
+                if (strTester.equals(strChecker) == true) {
+                  
+                    return true;
+                
+                } 
+            }
+    }
+
+    return false;
+
+    }
+
+
+    /** A method that finds the alphabetically first word and returns it. 
+    * @author N. Wan
+    * @param filenametxt The file that contains all the information needed.
+    * @return highestWord The word that is most alphabetically first.
+    * @IOExecption 
+    */
+    public static String alphaWord(String filenametxt) throws IOException { 
+     
+        // Initialize buffer reader and variables.
+        BufferedReader Br = new BufferedReader(new FileReader(filenametxt));
+        String strLine = Br.readLine();
+        String highestWord = Br.readLine();
+
+    // Read the lines in the file and compare them to see which word is the alphabetically first word. 
+    while(strLine != null) {
+        
+        strLine = Br.readLine();
+
+        if (strLine.compareTo(highestWord) > 0) {
```

Comment:
an error occurs on all tests for this `alphaWord` method because at some point it tries to compare a string to a null value. It's because your `readLine` occurs at the beginning of the loop body.